mlflow/pyfunc/cli.py

```diff
@@ -20,13 +61,22 @@ def commands():
 @cli_args.MODEL_PATH
 @cli_args.RUN_ID
 @click.option("--port", "-p", default=5000, help="Server port. [default: 5000]")
-def serve(model_path, run_id, port):
+@click.option("--no-conda", is_flag=True,
+              help="If specified, will assume that MLModel is running within a Conda environment "
+                   "with the necessary dependencies for the current project instead of attempting "
+                   "to create a new conda environment. Only valid if running locally.")
+def serve(model_path, run_id, port, no_conda):
     """
     Serve a PythonFunction model saved with MLflow.
 
     If a ``run_id`` is specified, ``model-path`` is treated as an artifact path within that run;
     otherwise it is treated as a local path.
     """
+    model_env_file = load_model_env(model_path, run_id)
+    if not no_conda and model_env_file is not None:
+        conda_env_path = os.path.join(model_path, model_env_file)
+        return _rerun_in_conda(conda_env_path)
```

Comment:
Could we add an eprint log message in this code path to indicate to the user that we're going to setup a conda env? In the `mlflow run` case, for example, we say

```
=== Creating conda environment mlflow-26f7fc828275cc3ad477d7f599003a7e136e2c8d ===
=== Running command 'source activate mlflow-26f7fc828275cc3ad477d7f599003a7e136e2c8d && python train.py 0.3 0.1' in run with ID 'ce480854c1e94249a6a3c787983a1991' ===
```

This is useful for the user to help debug if something weird happens.